src/gui/guiFormSpecMenu.cpp

```diff
@@ -2668,24 +2668,12 @@ void GUIFormSpecMenu::parseElement(parserData* data, const std::string &element)
 	if (parseVersionDirect(element))
 		return;
 
-	std::vector<std::string> parts = split(element,'[');
-
-	// ugly workaround to keep compatibility
-	if (parts.size() > 2) {
-		if (trim(parts[0]) == "image") {
-			for (unsigned int i=2;i< parts.size(); i++) {
-				parts[1] += "[" + parts[i];
-			}
-		}
-		else { return; }
-	}
-
-	if (parts.size() < 2) {
+	size_t pos = element.find('[');
+	if (pos == std::string::npos)
 		return;
-	}
 
-	std::string type = trim(parts[0]);
-	std::string description = trim(parts[1]);
+	std::string type = trim(element.substr(0, pos));
+	std::string description = trim(element.substr(pos+1));
```

Comment:
Why is there a `trim` for `description` btw.? This makes it impossible to eg. set a space as default value for a `field[]`.

Reply:
Ah, that's why I couldn't have newlines at the end of my vertlabels.  I think this should be removed.

Reply:
You are right, this should be the responsibility of element specific parse function to handle `description` part.